### DISCUSSION TITLE
SQLAlchemy

### DISCUSSION THREAD

**Discussion by jrcastro2** [inveniosoftware/docs-invenio-rdm#629] [Trainings Completed]
**Speakers:** @slint

### Description

![image](https://github.com/inveniosoftware/docs-invenio-rdm/assets/1814661/263947b7-f257-498d-bb41-3c2eac0de3f2)

- From SQL 🐖 to Python 🐍 and back again
- The life and death of engines, sessions, and transactions
- Common pitfalls and WTFs
- Bonus: the journey from v1 -> v2, and why you should care

**Comment by jrcastro2** [comment]
Indico event: https://indico.cern.ch/event/1446142/

**Comment by fenekku** [comment]
Hey @jrcastro2 is there a way for non-cern folks to get access to that presentation (slides / notes)? That and other trainings seem really interesting to the wider community. Thanks!

**Reply by slint** [reply]
This was a bit of an unconventional training I did, with some "live coding" elements. I've pushed the slides and code at https://github.com/slint/sqlalchemy-training

We should have probably recorded some version of this, though I'm happy if there's enough popular demand to do an online version of it.

Other talks I think have PDF slides, so we could upload them to the Indico events.

**Reply by fenekku** [reply]
Thank you so much :pray: . I've been adopting the new SQLAlchemy 2.0 for our internal InvenioRDM modules for a while, but I've also always had the hardest time figuring out what SQLAlchemy wants us to do for `session` / `flush` and how to relate to actual DB transactions (like head scratchers like https://github.com/inveniosoftware/invenio-rdm-records/blob/ee29707c26afa3caba6da968aae61c610acaf4a4/invenio_rdm_records/utils.py#L34 and other weird stuff). I haven't taken a look at the content yet, but I'd love any other resources on that (the official docs leave me more confused than when I started reading them).
